lib/ui/painting.dart

```diff
@@ -2027,8 +2027,8 @@ class Gradient extends Shader {
        assert(colors != null),
        assert(tileMode != null),
        super._() {
-    if (matrix4 != null && matrix4.length != 16)
-      throw new ArgumentError('"matrix4" must have 16 entries.');
+    if (matrix4 != null)
+      assert(_matrix4IsValid(matrix4));
```

Comment:
You can write this trivially more optimally as:

```dart
  assert(matrix4 == null || _matrix4IsValid(matrix4));
```

Reply:
Done.